embedx/embedx.go

```diff
@@ -0,0 +1,121 @@
+package embedx
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	"github.com/tidwall/gjson"
+
+	_ "embed"
+)
+
+//go:embed config.schema.json
+var ConfigSchema string
+
+//go:embed identity_meta.schema.json
+var IdentityMetaSchema string
+
+//go:embed identity_extension.schema.json
+var IdentityExtensionSchema string
+
+type SchemaType int
+type SchemaID string
+
+const (
+	Config SchemaType = iota
+	IdentityMeta
+	IdentityExtension
+)
+
+/*var (
+	ConfigSchemaID            SchemaID
+	IdentityMetaSchemaID      SchemaID
+	IdentityExtensionSchemaID SchemaID
+)*/
+
+type Schema struct {
+	id           string
+	data         string
+	dependencies []*Schema
+}
+
+var (
+	schemas = map[SchemaType]*Schema{
+		Config: {
+			id:           gjson.Get(ConfigSchema, "$id").Str,
+			data:         ConfigSchema,
+			dependencies: nil,
+		},
+		IdentityMeta: {
+			id:   gjson.Get(IdentityMetaSchema, "$id").Str,
+			data: IdentityMetaSchema,
+			dependencies: []*Schema{
+				{
+					id:           gjson.Get(IdentityExtensionSchema, "$id").Str,
+					data:         IdentityExtensionSchema,
+					dependencies: nil,
+				},
+			},
+		},
+		IdentityExtension: {
+			id:           gjson.Get(IdentityExtensionSchema, "$id").Str,
+			data:         IdentityExtensionSchema,
+			dependencies: nil,
+		},
+	}
+)
+
+func getSchema(schema SchemaType) (*Schema, error) {
+	if val, ok := schemas[schema]; ok {
+		return val, nil
+	}
+	return nil, fmt.Errorf("the specified schema type (%d) is not supported", int(schema))
+}
+
+func (s SchemaType) GetSchemaID() string {
+	return schemas[s].id
+}
+
+func (s SchemaID) ToString() string {
+	return string(s)
+}
```

Comment:
```suggestion
```
unused?

Reply:
ahh yeah forgot to take this out. I refactored a lot of code here